sdks/java/extensions/avro/build.gradle

```diff
@@ -63,67 +74,91 @@ dependencies {
   testImplementation library.java.junit
   testRuntimeOnly project(path: ":runners:direct-java", configuration: "shadow")
   testRuntimeOnly library.java.slf4j_jdk14
-  avroVersions.each {
-    "avroVersion$it.key" "org.apache.avro:avro:$it.value"
-    "avroVersion$it.key" "org.apache.avro:avro-tools:$it.value"
+  avroVersions.each { k,v ->
+    "avroVersion$k"(project(path: ":sdks:java:core", configuration: "shadowTest")) {
+      // Exclude Avro dependencies from "core" since Avro support moved to this extension
+      exclude group: "org.apache.avro", module: "avro"
+    }
+    "avroVersion$k" library.java.junit
+    "avroVersion$k" project(path: ":runners:direct-java", configuration: "shadow")
+    "avroVersion$k" library.java.slf4j_jdk14
+    "avroVersion$k" "org.tukaani:xz:1.9" // makes as provided since 1.9
+    "avroVersion$k" library.java.zstd_jni // makes as provided since 1.9
+    "avroVersion${k}Generate" "org.apache.avro:avro-tools:$v"
   }
 }
 
-avroVersions.each { kv ->
-  configurations."avroVersion$kv.key" {
-    resolutionStrategy {
-      force "org.apache.avro:avro:$kv.value"
+task printSourceSetInformation(){
+
+  doLast{
+    sourceSets.each { srcSet ->
+      println "["+srcSet.name+"]"
+      print "-->Source directories: "+srcSet.allJava.srcDirs+"\n"
+      print "-->Output directories: "+srcSet.output.classesDirs.files+"\n"
+      print "-->Compile classpath:\n"
+      srcSet.runtimeClasspath.files.each {
+        print "  "+it.path+"\n"
+      }
+      println ""
     }
   }
+}
+
+avroVersions.each { k, v ->
+  configurations."avroVersion$k" {
+    resolutionStrategy.force "org.apache.avro:avro:$v", "org.apache.avro:avro:$v:tests", library.java.joda_time
```

Comment:
Appending `joda_time` too as `force` is transitive and downgrades the lib when using `1.9`